maps_dashboards/public/components/layer_config/documents_config/document_layer_source.tsx

```diff
@@ -30,14 +30,12 @@ interface Props {
   setSelectedLayerConfig: Function;
   selectedLayerConfig: DocumentLayerSpecification;
   setIsUpdateDisabled: Function;
-  layersIndexPatterns: IndexPattern[];
 }
 
 export const DocumentLayerSource = ({
```

Comment:
It looks unnecessary to keep the component `DocumentLayerSource` state of `selectedNumber` `tooltipSelection`, etc. 

It could use layer config passed by `props` and update the layer config directly with `setSelectedLayerConfig`, like what we do in `onFiltersUpdated`. Layer config passed by props should be the only source of truth. What do you think?

Reply:
Yeah, just updated the code, thanks!